state/application.go

```diff
@@ -1167,6 +1167,16 @@ func (a *Application) SetCharm(cfg SetCharmConfig) (err error) {
 	if cfg.Charm.Meta().Subordinate != a.doc.Subordinate {
 		return errors.Errorf("cannot change an application's subordinacy")
 	}
+	currentCharm, err := a.st.Charm(a.doc.CharmURL)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if cfg.Charm.Meta().Deployment != currentCharm.Meta().Deployment {
+		if currentCharm.Meta().Deployment == nil ||
+			cfg.Charm.Meta().Deployment.DeploymentType != currentCharm.Meta().Deployment.DeploymentType {
```

Comment:
is it possible `cfg.Charm.Meta().Deployment == nil` ?

Reply:
No because of the first part of the if{} check tat cfg Meta != current Meta